playground/infrastructure/test_cd_helper.py

```diff
@@ -0,0 +1,138 @@
+    yield delete_temp_folder
+    if os.path.exists(TEMP_FOLDER):
+        shutil.rmtree(TEMP_FOLDER)
+
+
+def upload_blob():
+    """
+    Fake method for mocking
+    Returns: None
+
+    """
+    return None
+
+
+def test__get_cloud_file_name():
+    """
+    Test getting the path where file will be stored at the bucket
+    Returns:
+
+    """
+    expected_result = "SDK_JAVA/base_folder/file.java"
+    expected_result_with_extension = "SDK_JAVA/base_folder/file.output"
+    assert CDHelper()._get_cloud_file_name(SDK_JAVA, "base_folder", "file") == expected_result
+    assert CDHelper()._get_cloud_file_name(SDK_JAVA, "base_folder", "file", "output") == expected_result_with_extension
+
+
+def test__get_data_from_template():
+    """
+    Test finding beam-playground tag and extract the information of the example to dictionary
+    Returns:
+
+    """
+    code = '''
+    Some text
+    
+    /*
+    Beam-playground:
+      name: name
+      description: description
+      multifile: false
+      categories:
+        - category-1
+        - category-2
+    */
+    
+    More of some text
+    '''
+    expected_result = {'name': 'name', 'description': 'description', 'multifile': False,
+                       'categories': ['category-1', 'category-2']}
+    assert CDHelper()._get_data_from_template(code) == expected_result
+
+
+def test__write_to_os(delete_temp_folder):
+    """
+    Test writing code of an example, output and meta info to the filesystem (in temp folder)
+    Args:
+        delete_temp_folder:
+
+    Returns:
+
+    """
+    example = Example("name", "pipeline_id", SDK_JAVA, "filepath", "code_of_example",
+                      "output_of_example", STATUS_UNSPECIFIED)
+    object_meta = {'name': 'name', 'description': 'description', 'multifile': False,
+                   'categories': ['category-1', 'category-2']}
+    expected_result = {'SDK_JAVA/name/name.java': 'temp/pipeline_id/SDK_JAVA/name/name.java',
+                       'SDK_JAVA/name/name.output': 'temp/pipeline_id/SDK_JAVA/name/name.output',
+                       'SDK_JAVA/name/meta.info': 'temp/pipeline_id/SDK_JAVA/name/meta.info'}
+    assert CDHelper()._write_to_os(example, object_meta) == expected_result
+
+    object_meta.pop('name')
+    with pytest.raises(KeyError):
+        CDHelper()._write_to_os(example, object_meta)  # "object_meta should contain 'name' field"
+
+
+def test__save_to_cloud_storage(mocker):
+    """
+    Test saving examples, outputs and meta to bucket
+    Args:
+        mocker:
+
+    Returns:
+
+    """
+    mocker.patch(
+        'cd_helper.CDHelper._upload_blob',
+        return_value=upload_blob
+    )
+    mocker.patch(
+        'cd_helper.CDHelper._write_to_os',
+        return_value={"": ""}
+    )
+    example = Example("name", "pipeline_id", SDK_JAVA, "filepath", "code_of_example",
+                      "output_of_example", STATUS_UNSPECIFIED)
+
+    CDHelper()._save_to_cloud_storage([example])
```

Comment:
Can we at least check here that it called the functions we expect it to call?

Reply:
good suggestion

Reply:
Please, don't resolve conversations by yourself – they need to be resolved by the reviewer to track that the changes are made, I almost missed this one😅
Could you please add checks here that it called the functions we expect it to call?

Reply:
This one I haven't resolved :)